graphql/exec.go

```diff
@@ -21,12 +21,75 @@ func CollectFields(doc *query.Document, selSet []query.Selection, satisfies []st
 	return collectFields(doc, selSet, satisfies, variables, map[string]bool{})
 }
 
+func runDirectives(directives common.DirectiveList, variables map[string]interface{}) bool {
+	for _, directive := range directives {
+		if !runSingleDirective(directive, variables) {
+			return false
+		}
+	}
+	return true
+}
+
+func runSingleDirective(directive *common.Directive, variables map[string]interface{}) bool {
+	arg := directive.Args.MustGet("if")
+	fieldAllowed := true
+	switch arg.(type) {
+	case *common.Variable:
+		{
+			value := arg.Value(variables)
+			if realValue, ok := value.(bool); ok {
+				fieldAllowed = realValue
+			}
+		}
+	case *common.BasicLit:
+		{
+			value := arg.Value(nil)
+			if realValue, ok := value.(bool); ok {
+				fieldAllowed = realValue
+			}
+		}
+	}
+
+	switch directive.Name.Name {
+	case "skip":
+		{
+			return !fieldAllowed
+		}
+	case "include":
+		{
+			return fieldAllowed
+		}
+	default:
+		// Custom directives are not supported yet
+		return fieldAllowed
+	}
+}
+
+func populateVariables(operation *query.Operation, variables map[string]interface{}) map[string]interface{} {
+	if variables == nil {
+		variables = make(map[string]interface{})
+	}
+	for _, variable := range operation.Vars {
```

Comment:
This seems like a lot of work to do on every fieldset, I think this goes away if we pass context in?

Reply:
If there will be one specific place which will have both variables and their default values, this will be super good!

Reply:
You can do this check inline, the one place you need the value from a variable, this function shouldn't exist.

And it really shouldn't mutate `variables`, they should be static after the initial validation pass.